src/helpers.c

```diff
@@ -595,6 +595,32 @@ int swupd_init(void)
 	return ret;
 }
 
+/* The strtol function is commonly used to convert a string to a number and
+ * the result is frequently stored in an int type. Type casting a long to an
+ * int can cause overflows. This function sets 'err' to a nonzero value when
+ * either the strtol return value or the long to int type conversion overflows.
+*/
+int strtoi_err(const char *str, char **endptr, int base, int *err)
```

Comment:
We are always using base 10 and I don't see a reason to use other bases in any other case we are handling, so you can take the base out of the arguments list.
I prefer to always return the error and use the parameter for the value you wan't to retrieve. This is particularly good for this case because if you use incorrect int sizes the compiler will warn you.

So if you have

    int strtoi_err(const char *str, char **endptr, int *value);

and use

    char value;
    err = strtoi_err("1", NULL, &value);

The compiler will warn you.